src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -277,16 +277,26 @@ private Map<Integer, Float> doANNSearch(final LeafReaderContext context, final B
                 throw new RuntimeException("Index has already been closed");
             }
             int[] parentIds = getParentIdsArray(context);
-            results = JNIService.queryIndex(
-                indexAllocation.getMemoryAddress(),
-                knnQuery.getQueryVector(),
-                knnQuery.getK(),
-                knnEngine,
-                filterIds,
-                filterType.getValue(),
-                parentIds
-            );
-
+            if (knnQuery.getK() > 0) {
+                results = JNIService.queryIndex(
+                    indexAllocation.getMemoryAddress(),
+                    knnQuery.getQueryVector(),
+                    knnQuery.getK(),
+                    knnEngine,
+                    filterIds,
+                    filterType.getValue(),
+                    parentIds
+                );
+            } else {
+                int indexMaxResultWindow = KNNSettings.getIndexMaxResultWindow(knnQuery.getIndexName());
+                results = JNIService.radiusQueryIndex(
+                    indexAllocation.getMemoryAddress(),
+                    knnQuery.getQueryVector(),
+                    knnQuery.getRadius(),
```

Comment:
Do we know this is set?

Reply:
Yes, before we do the ANNSearch, the query parameters already will be verified in KNNQueryBuilder step https://github.com/opensearch-project/k-NN/blob/36a2dd670ed75cef501232ac32e26853bbcc064a/src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java#L90-L120